doc/multi_asic/SONiC_multi_asic_hld.md

```diff
@@ -0,0 +1,937 @@
+
+A linux network namespace is created for every ASIC that's controlled by the same CPU.
+This creates a separate linux network stack including routes and network devices for every ASIC.
+The interfaces for a given ASIC is linked to its corresponding namespace.
+
+In a multi-ASIC system, very commonly the ASICs are physically connected in a clos fabric topology. With sonic container dockers running as separate namespaces for each ASIC in a multi-ASIC system, we can model and configure the system as if there is a spine-leaf network topology within the box.
+
+![Archtecture Diagram](images/architecture_diagram.jpg)
+
+A high level software architectural view for a multi-asic system with 6 ASIC is shown as follows:
+
+
+
+![Archtecture Detailed View](images/multi_asic_architecture.jpg)
+
+
+
+### 2.3.1. ASIC roles
+
+On a multi ASIC system, each ASIC can act as 
+ - Frontend ASIC, which connects to external devices. These ASICs have some ports as frontpanel ports and some as internal ports.
+ - Backend ASIC, which have only internal links. These ASICs only connect to other Frontend ASICs.  In a chassis system, the fabric ASIC is considered as a Backend ASIC.
+
+### 2.3.2. Control Plane
+ Within a Multi ASIC SONiC system, internal control plane can be setup in following ways
+
+#### 2.3.2.1. BGP on backend ASICs
+ In this approach:
+ - BGP instance is running on all the ASICs.
+ - iBGP sessions are formed between each frontend and backend ASIC.
+ - Route reflector is configured on the backend ASICs
+ - All the ASICs have the same view of the network.
+
+![BGP on backend ASICs](images/bgp_on_backend_asic.jpg)
+
+ In the above picture,
+
+  - ASIC0 and ASIC1 have ibgp sessions established with the backend ASIC, ASIC2.
+  - Router A and Router B establish eBGP sessions with ASIC0 and ASIC1 respectively
+  - Let say a prefix, _192.168.199.193_, is learnt from Router A on ASIC0.
+  ```
+    root@sonic:/home/admin# ip -n asic0 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.106.0.1  dev PortChannel1002 weight 1
+  ```
+  - ASIC0 advertises this prefix to ASIC2. On ASIC2 the nexthop is the interface connected to ASIC0
+```
+    root@sonic:/home/admin# ip -n asic2 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.0.107.1  dev PortChannel4009 weight 1
+
+```
+- ASIC1 learns these routes from ASIC2 as route reflection is configured on ASIC2
+```
+    root@sonic:/home/admin# ip -n asic1 route show  192.168.199.193
+    192.168.199.193 proto 186 src 172.16.132.64 metric 20
+        nexthop via 10.0.107.12  dev PortChannel4007 weight 1
+```
+  - A packet with destination ip as 192.168.199.193

```

Comment:
confused here, could you review this part and make the name/ip/asicX correct?  Looks the description and the picture are not matching.